### DISCUSSION TITLE
I want to get  external cluster credentials after "argocd cluster add".What should i do？

### DISCUSSION THREAD

**Discussion by breakALeg111** [argoproj/argo-cd#18274] [Q&A]
Steps：
    1、I add a cluster by “argocd cluster add”
    2、A secret named “cluster-kubernetes.xxx.904396332” are created in the specified namespace. There are a json which contains information about "caData、keyData、certData" in data.config.
    3、I call http://xxx.xxx.xxx.xxx/api/v1/clusters. But only caData and certData are returned.
If i want to connect external cluster and do something. I also need keyData. How can i get keyData？

If calling the API interface is not feasible, how can I operate an external cluster managed by argocd while only knowing argocd's username and password？If you have any good suggestions, thank you very much！

**Comment by agaudreault** [comment]
You cannot. ArgoCD cannot be leveraged to obtain the cluster credentials. If you are doing `argocd cluster add` locally, that means you already have the authentication information that you provide to Argo to authenticate to that cluster. You should use that information to query the cluster.
